File: excerbuys-app/lib/types/activity.dart
import 'package:excerbuys/types/general.dart';
import 'package:flutter/material.dart';

enum STEPS_AGGREGATION_TYPE { HOURLY, DAILY, MONTHLY }

enum STEPS_AGGREGATION_VALUE { TOTAL, MEAN }

enum ACTIVITY_TYPE { WALKING, RUNNING, BIKE_RIDING, SWIMMING }

class ActivityMetadata {
  final String icon;
  final String name;
  final Color color;
  final Color? textColor;

  const ActivityMetadata(
      {required this.icon,
      required this.name,
      required this.color,
      this.textColor});
}

class ITrainingEntry {
  final String uuid;
  final int points;
  final String type;
  final String userId;
  final int duration;
  final DateTime createdAt;
  final int? calories;
  final int? distance;

  const ITrainingEntry(
      {required this.uuid,
      required this.points,
      required this.duration,
      this.calories,
      this.distance,
      required this.type,
      required this.userId,
      required this.createdAt});

  String toJson() {
    return """{
        "uuid":"${uuid}",
        "points":"${points}",
        "type":"${type}",
        "user_id":"${userId}",
        "duration":"${duration}",
        "calories":"${calories}",
        "distance":"${distance}",
        "created_at":"${createdAt}"
      }""";
  }
}

class IHourlyStepsEntry {
  final int total;
  final String userId;
  final DateTime timestamp;
  final String uuid;

  const IHourlyStepsEntry(
      {required this.uuid,
      required this.total,
      required this.userId,
      required this.timestamp});

  String toJson() {
    return """{
        "uuid":"$uuid",
        "timestamp":"$timestamp",
        "user_id":"$userId",
        "total":"$total"
      }""";
  }
}

typedef IStoreStepsData = Map<DateTime, Map<int, int>>;

File: server/shared/types/activity.ts
export interface ITrainingEntry {
  points: number;
  duration: number;
  calories?: number;
  distance?: number;
  type: string;
  user_id: string;
  uuid: number;
  created_at?: string;
}

export interface ITrainingEntryResponse {
  id: number;
  created_at: string;
  points: number;
  duration: number;
  calories?: number;
  distance?: number;
  type: string;
}

export interface IHourlyStepEntry {
  uuid: string;
  timestamp: string;
  total: number;
  user_id: string;
}

export interface IDailyStepEntry {
  uuid: string;
  timestamp: string;
  total: number;
  user_id: string;
  mean: number;
}
